proxy/src/main/java/org/apache/rocketmq/proxy/service/mqclient/MQClientAPIFactory.java

```diff
@@ -55,6 +55,9 @@ protected void init() {
         System.setProperty(ClientConfig.SEND_MESSAGE_WITH_VIP_CHANNEL_PROPERTY, "false");
         ProxyConfig proxyConfig = ConfigurationManager.getProxyConfig();
         if (StringUtils.isEmpty(proxyConfig.getNamesrvDomain())) {
+            if (proxyConfig.getNamesrvAddr() == null || proxyConfig.getNamesrvAddr().isEmpty()) {
```

Comment:
Try [isNullOrEmpty](https://guava.dev/releases/23.0/api/docs/com/google/common/base/Strings.html#isNullOrEmpty-java.lang.String-) to simplify this line.